File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
       int low=0,mid=0,high=nums.length-1;
       while(mid<=high){
        if(nums[mid]==0){
            int temp=nums[low];
            nums[low]=nums[mid];
            nums[mid]=temp;
            mid=mid+1;
            low=low+1;
        }
        else if(nums[mid]==1){
            mid=mid+1;
        }
        else{
            int temp=nums[mid];
            nums[mid]=nums[high];
            nums[high]=temp;
            high=high-1;
        }
       } 
    }
}

File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int i=0,j=0,k=nums.size()-1;
        while(j<=k){
            if(nums[j]==0){
                int temp1=nums[j];
                nums[j]=nums[i];
                nums[i]=temp1;
                j++;
                i++;
            }
            else if(nums[j]==2){
                int temp=nums[j];
                nums[j]=nums[k];
                nums[k]=temp;
                // j++;
                k--;
            }
            else{
                j++;
            }
            //cout << nums[j];
        }
    }
};